.github/workflows/recreate-macos-runners.yml

```diff
@@ -0,0 +1,137 @@
+name: Recreate macOS self-hosted runners on schedule
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Once each 24 hours, at 1 during the night
+    - cron: "0 1 * * *"
+
+concurrency:
+  group: recreate-macos-runners
+  cancel-in-progress: true
+
+jobs:
+  recreate:
+    if: github.repository == 'Homebrew/homebrew-core'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - runner_name: monterey-1
+            label: 12
+            vm_name: monterey
+            node: macpro-4
+            port: 8822
+          - runner_name: monterey-2
+            label: 12
+            vm_name: monterey
+            node: macpro-4
+            port: 8823
+          - runner_name: monterey-3
+            label: 12
+            vm_name: monterey
+            node: macpro-4
+            port: 8824
+          - runner_name: monterey-4
+            label: 12
+            vm_name: monterey
+            node: macpro-4
+            port: 8825
+          - runner_name: bigsur-1
+            label: 11
+            vm_name: bigsur
+            node: macpro-5
+            port: 8822
+          - runner_name: bigsur-2
+            label: 11
+            vm_name: bigsur
+            node: macpro-5
+            port: 8823
+          - runner_name: bigsur-3
+            label: 11
+            vm_name: bigsur
+            node: macpro-5
+            port: 8824
+          - runner_name: bigsur-4
+            label: 11
+            vm_name: bigsur
+            node: macpro-5
+            port: 8825
+          - runner_name: catalina-1
+            label: 10.15
+            vm_name: catalina
+            node: macpro-6
+            port: 8822
+          - runner_name: catalina-2
+            label: 10.15
+            vm_name: catalina
+            node: macpro-6
+            port: 8823
+          - runner_name: catalina-3
+            label: 10.15
+            vm_name: catalina
+            node: macpro-6
+            port: 8824
+          - runner_name: catalina-4
+            label: 10.15
+            vm_name: catalina
+            node: macpro-6
+            port: 8825
+    steps:
+      - name: Install openconnect
+        run: brew install openconnect
+
+      - name: Checkout Orka API Client gem
+        uses: actions/checkout@v2
+        with:
+          repository: Homebrew/orka_api_client
+          path: orka_api_client
+          persist-credentials: false
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.1"
+          bundler-cache: true
+          working-directory: orka_api_client
+
+      - name: Install Orka API Client gem
+        working-directory: orka_api_client
+        run: |
+          bundle exec rake
+          gem install pkg/orka_api_client-*.gem
```

Comment:
This would be nice to turn into a dedicated formula and/or action.

Reply:
This can be simplified if/when we publish it to RubyGems.